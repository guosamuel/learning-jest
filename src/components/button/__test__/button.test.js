import React from 'react'
import ReactDOM from 'react-dom'
import Button from './../button'

import { render } from '@testing-library/react'

import renderer from "react-test-renderer"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<Button></Button>, div)
})

it("renders button correctly", () => {
  const { getByTestId } = render(<Button label="click me please"></Button>)
  expect(getByTestId('button')).toHaveTextContent("click me please")
})

it("renders button correctly", () => {
  const { getByTestId } = render(<Button label="save"></Button>)
  expect(getByTestId('button')).toHaveTextContent("save")
})

it("matches snapshot alpha", () => {
  const tree = renderer.create(<Button label="save"></Button>).toJSON()
  expect(tree).toMatchSnapshot()
})

it("matches snapshot beta", () => {
  const tree = renderer.create(<Button label="click me please"></Button>).toJSON()
  expect(tree).toMatchSnapshot()
})
