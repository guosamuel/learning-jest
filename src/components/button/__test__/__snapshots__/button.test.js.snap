// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot alpha 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  save
</div>
`;

exports[`matches snapshot beta 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  click me please
</div>
`;
